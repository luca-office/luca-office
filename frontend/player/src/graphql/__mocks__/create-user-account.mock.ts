import {CreateUserAccountMutation_createUserAccount} from "../generated/CreateUserAccountMutation"
import {Salutation} from "../generated/globalTypes"

export const createUserAccountMock: CreateUserAccountMutation_createUserAccount = {
  id: "123",
  createdAt: new Date(2020, 12, 12).toISOString(),
  modifiedAt: new Date(2020, 12, 12).toISOString(),
  email: "test@test.de",
  firstName: "test user",
  lastName: "user",
  organization: "Cap3",
  __typename: "UserAccount",
  salutation: Salutation.Mrs,
  mayAdministrateUserAccounts: false,
  mayArchive: false,
  mayFinalizeWithoutPublishing: false,
  mayAdministrateRScripts: false
}
