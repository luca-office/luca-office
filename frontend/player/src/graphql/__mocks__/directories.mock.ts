import {DirectoryFragment} from "../generated/DirectoryFragment"

export const directoriesMock: DirectoryFragment[] = [
  {
    __typename: "Directory",
    id: "b6897871-96be-4927-894d-ee99b4a8f30a",
    createdAt: "2020-10-22T11:16:01.973Z",
    modifiedAt: "2020-10-22T11:16:01.973Z",
    name: "Bilder",
    parentDirectoryId: null,
    scenarioId: "4d09512c-37e2-4758-a457-ff9fa7338e16",
    sampleCompanyId: null
  },
  {
    __typename: "Directory",
    id: "814c5305-3b5c-463c-8012-e6bf1b5a1581",
    createdAt: "2020-10-22T11:16:18.621Z",
    modifiedAt: "2020-10-22T11:16:18.621Z",
    name: "Rechnungen",
    parentDirectoryId: null,
    scenarioId: "4d09512c-37e2-4758-a457-ff9fa7338e16",
    sampleCompanyId: null
  },
  {
    __typename: "Directory",
    id: "a90f0a95-1e7f-4486-becf-0d8ea02a1378",
    createdAt: "2020-10-22T11:16:29.558Z",
    modifiedAt: "2020-10-22T11:16:29.558Z",
    name: "Sonstiges",
    parentDirectoryId: null,
    scenarioId: "4d09512c-37e2-4758-a457-ff9fa7338e16",
    sampleCompanyId: null
  },
  {
    __typename: "Directory",
    id: "6b183a2e-0af6-4006-95b5-e2daca778616",
    createdAt: "2020-10-22T11:21:22.26Z",
    modifiedAt: "2020-10-22T11:21:22.26Z",
    name: "Urlaub",
    parentDirectoryId: "b6897871-96be-4927-894d-ee99b4a8f30a",
    scenarioId: "4d09512c-37e2-4758-a457-ff9fa7338e16",
    sampleCompanyId: null
  },
  {
    __typename: "Directory",
    id: "51cb5f14-19c2-45e7-a7e0-01501aec1f59",
    createdAt: "2020-10-22T11:21:28.455Z",
    modifiedAt: "2020-10-22T11:21:28.455Z",
    name: "Arbeit",
    parentDirectoryId: "b6897871-96be-4927-894d-ee99b4a8f30a",
    scenarioId: "4d09512c-37e2-4758-a457-ff9fa7338e16",
    sampleCompanyId: null
  },
  {
    __typename: "Directory",
    id: "a7fad23d-28ed-442a-8ff2-89b1a11addeb",
    createdAt: "2020-10-22T11:22:04.718Z",
    modifiedAt: "2020-10-22T11:22:04.718Z",
    name: "November",
    parentDirectoryId: "814c5305-3b5c-463c-8012-e6bf1b5a1581",
    scenarioId: "4d09512c-37e2-4758-a457-ff9fa7338e16",
    sampleCompanyId: null
  },
  {
    __typename: "Directory",
    id: "6ea48be2-1e65-44d5-be6a-cde64c2bd874",
    createdAt: "2020-10-22T11:22:09.724Z",
    modifiedAt: "2020-10-22T11:22:09.725Z",
    name: "Dezember",
    parentDirectoryId: "814c5305-3b5c-463c-8012-e6bf1b5a1581",
    scenarioId: "4d09512c-37e2-4758-a457-ff9fa7338e16",
    sampleCompanyId: null
  }
]
