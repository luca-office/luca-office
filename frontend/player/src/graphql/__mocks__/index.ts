export * from "./create-user-account.mock"
export * from "./scenarios.mock"
export * from "./survey-invitation.mock"
