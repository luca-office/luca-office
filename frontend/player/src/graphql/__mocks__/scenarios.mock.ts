import {Salutation} from "shared/graphql/generated/globalTypes"
import {Scenario} from "shared/models"

export const scenariosMock: Scenario[] = [
  {
    __typename: "Scenario",
    id: "18de306c-a8f8-4311-9a2f-ca5418a245ac",
    name: "Scenario 1",
    description: "description",
    date: new Date(2021, 12, 1).toISOString(),
    maxDurationInSeconds: 5400,
    introductionEmailId: null,
    authorId: "psdfoj-sdöfois-sdf",
    completionEmailAddress: "test@test.de",
    shouldDisplayTime: false,
    shouldHideReferenceBookChapters: false,
    sampleCompany: null,
    sampleCompanyId: null,
    tags: [],
    createdAt: new Date(2021, 12, 1).toISOString(),
    modifiedAt: new Date(2021, 12, 1).toISOString(),
    finalizedAt: null,
    publishedAt: null,
    archivedAt: null,
    introductionEmail: null,
    codingModel: null,
    author: {
      __typename: "UserAccount",
      firstName: "Max",
      lastName: "Mustermann",
      id: "psdfoj-sdöfois-sdf",
      createdAt: new Date(2021, 12, 1).toISOString(),
      modifiedAt: new Date(2021, 12, 1).toISOString(),
      email: "max.mustermann@muster.de",
      organization: "Muster GmbH",
      salutation: Salutation.Mr,
      mayAdministrateUserAccounts: false,
      mayArchive: false,
      mayFinalizeWithoutPublishing: false,
      mayAdministrateRScripts: false
    }
  },
  {
    __typename: "Scenario",
    id: "18de306c-a8f8-4311-9a2f-dfdf",
    name: "Scenario 2",
    description: "description",
    date: "2020-08-03 06:16:28.504000 +00:00",
    maxDurationInSeconds: null,
    introductionEmailId: null,
    authorId: "psdfoj-sdöfois-sddsf",
    completionEmailAddress: "test@test.de",
    sampleCompany: null,
    sampleCompanyId: null,
    shouldDisplayTime: false,
    shouldHideReferenceBookChapters: false,
    tags: [],
    createdAt: new Date(2021, 12, 1).toISOString(),
    modifiedAt: new Date(2021, 12, 1).toISOString(),
    finalizedAt: null,
    publishedAt: null,
    archivedAt: null,
    introductionEmail: null,
    codingModel: null,
    author: {
      __typename: "UserAccount",
      firstName: "Max",
      lastName: "Mustermann",
      id: "psdfoj-sdöfois-sddsf",
      createdAt: new Date(2021, 12, 1).toISOString(),
      modifiedAt: new Date(2021, 12, 1).toISOString(),
      email: "max.mustermann@muster.de",
      organization: "Muster GmbH",
      salutation: Salutation.Mr,
      mayAdministrateUserAccounts: false,
      mayArchive: false,
      mayFinalizeWithoutPublishing: false,
      mayAdministrateRScripts: false
    }
  }
]
