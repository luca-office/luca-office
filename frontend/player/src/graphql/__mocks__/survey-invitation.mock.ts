import {
  participantDataMock,
  participantProjectModuleProgressesMock,
  surveyIdMock,
  surveysMock,
  userAccountMock
} from "shared/graphql/__mocks__"
import {SurveyInvitation} from "shared/models"

const survey = {...surveysMock[0], id: surveyIdMock}

export const surveyInvitationMock: SurveyInvitation = {
  __typename: "SurveyInvitation",
  id: "8a49147d-3c8b-4995-95bf-ab4ab9299791",
  isOpenParticipation: false,
  survey,
  token: "sdfdss",
  userAccountId: userAccountMock.id,
  userAccount: userAccountMock,
  createdAt: new Date(2020, 10, 5).toISOString(),
  modifiedAt: new Date(2020, 10, 15).toISOString(),
  email: "info@cap3.de",
  surveyId: surveyIdMock,
  participantData: participantDataMock[0],
  projectModuleProgresses: participantProjectModuleProgressesMock
}
