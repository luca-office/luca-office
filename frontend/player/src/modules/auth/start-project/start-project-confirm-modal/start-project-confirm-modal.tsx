import {css} from "@emotion/react"
import React from "react"
import {Heading, Modal, Text} from "shared/components"
import {HeadingLevel} from "shared/enums"
import {SurveyInvitation} from "shared/models"
import {Flex, FontWeight, spacingMedium, spacingSmall, TextSize} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {convertSecondsToMinutes} from "shared/utils"

interface Props {
  readonly onConfirm: () => void
  readonly onDismiss: () => void
  readonly surveyInvitation: SurveyInvitation
}

export const StartProjectConfirmModal: React.FC<Props> = ({onConfirm, onDismiss, surveyInvitation}) => {
  const {t} = useLucaTranslation()
  return (
    <Modal
      onConfirm={onConfirm}
      onDismiss={onDismiss}
      customStyles={styles.modal}
      title={`${t("project__start_dialog_title", {title: surveyInvitation.survey.project.name})}`}
      confirmButtonKey="project__start_confirm"
      dismissButtonKey="back_button">
      <Text>
        {t("project__start_dialog_text_confirm_modal", {
          durationInMinutes: convertSecondsToMinutes(surveyInvitation.survey.project.maxDurationInSeconds)
        })}
      </Text>
      <div css={styles.tokenRow}>
        <Heading fontWeight={FontWeight.Bold} level={HeadingLevel.h3}>
          {t("project__start_dialog_text_confirm_modal_code")}
        </Heading>
        <Text customStyles={styles.token} size={TextSize.Medium}>
          {surveyInvitation.token}
        </Text>
      </div>

      <Text customStyles={styles.confirmText}>{t("project__start_dialog_text_2")}</Text>
    </Modal>
  )
}

const styles = {
  modal: css({
    width: "40vw"
  }),
  tokenRow: css(Flex.row, {
    alignItems: "baseline",
    marginTop: spacingMedium,
    marginBottom: spacingMedium
  }),
  token: css({
    marginLeft: spacingSmall
  }),
  confirmText: css({
    marginBottom: spacingMedium
  })
}
