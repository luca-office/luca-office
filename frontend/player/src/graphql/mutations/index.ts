import * as createUserAccount from "./create-user-account.graphql"
import * as startOpenParticipation from "./start-open-participation.graphql"
import * as userAccountSurveyParticipation from "./user-account-survey-participation.graphql"

export const createUserAccountMutation = createUserAccount
export const startOpenParticipationMutation = startOpenParticipation
export const userAccountSurveyParticipationMutation = userAccountSurveyParticipation
