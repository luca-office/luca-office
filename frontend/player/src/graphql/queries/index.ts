import * as filesForSampleCompany from "shared/graphql/queries/file/files-for-sample-company.graphql"
import * as filesForScenario from "shared/graphql/queries/file/files-for-scenario.graphql"
import * as project from "./project.graphql"
import * as scenarioQuestionnaires from "./questionnaire/scenario-questionnaires.graphql"
import * as scenario from "./scenario.graphql"
import * as surveyInvitation from "./survey-invitation.graphql"

export const filesForScenarioQuery = filesForScenario
export const filesForSampleCompanyQuery = filesForSampleCompany
export const scenarioQuery = scenario
export const surveyInvitationQuery = surveyInvitation
export const scenarioQuestionnairesQuery = scenarioQuestionnaires
export const projectQuery = project
