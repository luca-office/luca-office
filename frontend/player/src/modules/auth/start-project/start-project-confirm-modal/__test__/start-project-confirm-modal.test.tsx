import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Heading, Text} from "shared/components"
import {surveyInvitationMock} from "../../../../../graphql/__mocks__"
import {StartProjectConfirmModal} from "../start-project-confirm-modal"

describe("Start-Project-Confirm-Modal", () => {
  it("renders correctly", () => {
    const component = create(
      <StartProjectConfirmModal onDismiss={jest.fn()} onConfirm={jest.fn()} surveyInvitation={surveyInvitationMock} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure", () => {
    const component = shallow(
      <StartProjectConfirmModal onDismiss={jest.fn()} onConfirm={jest.fn()} surveyInvitation={surveyInvitationMock} />
    )
    expect(component.find(Text)).toHaveLength(3)
    expect(component.find(Heading)).toHaveLength(1)
  })
})
