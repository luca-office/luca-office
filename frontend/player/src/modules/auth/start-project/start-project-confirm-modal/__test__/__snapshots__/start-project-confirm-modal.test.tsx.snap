// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start-Project-Confirm-Modal renders correctly 1`] = `
<div
  className="css-1iw6hgd-Overlay"
>
  <div>
    <div
      className="css-uexrgo-Modal"
    >
      <header
        className="css-14sl779-Modal"
      >
        <div
          className="css-1f6p2vj-Heading"
        >
          project__start_dialog_title
        </div>
        <div
          className="css-12vbzj8-Icon"
          dangerouslySetInnerHTML={
            {
              "__html": {},
            }
          }
          onClick={[MockFunction]}
        />
      </header>
      <section
        className="css-1gs9x9w-Modal"
      >
        <div
          className="css-fmthw-Text"
        >
          project__start_dialog_text_confirm_modal
        </div>
        <div
          className="css-2gm1mh-StartProjectConfirmModal"
        >
          <div
            className="css-8050ac-Heading"
          >
            project__start_dialog_text_confirm_modal_code
          </div>
          <div
            className="css-171ncof-Text"
          >
            sdfdss
          </div>
        </div>
        <div
          className="css-1p9qd62-Text"
        >
          project__start_dialog_text_2
        </div>
      </section>
      <footer
        className="css-ukdqb3-Modal"
      >
        <button
          className="cancel-button css-a1i4d5-Button"
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="css-95j2hm-Button"
          >
            back_button
          </div>
        </button>
        <button
          className="confirm-button css-1ex7tyh-Button"
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="css-95j2hm-Button"
          >
            project__start_confirm
          </div>
        </button>
      </footer>
    </div>
  </div>
</div>
`;
